grpc-client-server/prober_client/main.go

```diff
@@ -4,19 +4,55 @@ package main
 import (
 	"context"
 	"flag"
+	"fmt"
+	"io"
 	"log"
+	"os"
+	"strings"
+	"sync"
+	"time"
 
-	pb "github.com/CodeYourFuture/immersive-go-course/grpc-client-server/prober"
+	pb "github.com/Berkeli/immersive-go/grpc-client-server/prober"
+	"github.com/jedib0t/go-pretty/table"
+	"github.com/schollz/progressbar/v3"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 )
 
+type Result struct {
+	Endpoint string
+	Failed   int32
+	Average  float32
+	Err      error
+}
+
+type ArrayFlag []string
+
+func (i *ArrayFlag) String() string {
+	return strings.Join(*i, ", ")
+}
+
+func (i *ArrayFlag) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
 var (
-	addr = flag.String("addr", "localhost:50051", "the address to connect to")
+	endpoints   = ArrayFlag{}
+	addr        = flag.String("addr", "localhost:50051", "the address to connect to")
+	nOfRequests = flag.Int("tries", 1, "number of requests to make")
+	timeout     = flag.Int("timeout", 3, "timeout in seconds, how long should we allow for probing")
```

Comment:
int + seconds works for duration flags - there's also [`flag.Duration`](https://pkg.go.dev/flag#Duration) which lets people pass more rich values